java-frontend/src/main/java/org/sonar/java/bytecode/cfg/BytecodeCFGBuilder.java

```diff
@@ -0,0 +1,235 @@
+    try(InputStream is = classLoader.getResourceAsStream(Convert.bytecodeName(((JavaSymbol.TypeJavaSymbol) methodSymbol.owner()).getFullyQualifiedName()) + ".class")) {
+      byte[] bytes = ByteStreams.toByteArray(is);
+      // to read bytecode with ASM not supporting Java 9, we will set major version to Java 8
+      if (Java9Support.isJava9Class(bytes)) {
+        Java9Support.setJava8MajorVersion(bytes);
+      }
+      ClassReader cr = new ClassReader(bytes);
+      BytecodeCFGMethodVisitor methodVisitor = new BytecodeCFGMethodVisitor(methodSymbol);
+      cr.accept(new ClassVisitor(Opcodes.ASM5) {
+        @Override
+        public MethodVisitor visitMethod(int access, String name, String desc, String signature, String[] exceptions) {
+          // TODO : name matching is not sufficient in case of overloading.
+          if (name.equals(methodSymbol.name())) {
+            return methodVisitor;
+          }
+          return null;
+        }
+      }, ClassReader.SKIP_DEBUG | ClassReader.SKIP_FRAMES);
+      return methodVisitor.cfg;
+    } catch (IOException e) {
+      throw Throwables.propagate(e);
+    }
+  }
+
+  static class BytecodeCFG {
+    List<Block> blocks;
+    Symbol.MethodSymbol methodSymbol;
+
+    BytecodeCFG(Symbol.MethodSymbol methodSymbol) {
+      this.methodSymbol = methodSymbol;
+      blocks = new ArrayList<>();
+      // create exit block
+      blocks.add(new Block(this));
+    }
+  }
+  static class Block {
+    int id;
+    BytecodeCFG cfg;
+    List<Instruction> instructions;
+    List<Block> successors;
+
+    Block(BytecodeCFG cfg) {
+      this.cfg = cfg;
+      this.id = cfg.blocks.size();
+      instructions = new ArrayList<>();
+      successors = new ArrayList<>();
+    }
+
+    void addInsn(int opcode) {
+      instructions.add(new Instruction(opcode));
+    }
+
+    Block createSuccessor() {
+      Block newBlock = new Block(cfg);
+      successors.add(newBlock);
+      cfg.blocks.add(newBlock);
+      return newBlock;
+    }
+
+    String printBlock() {
+      StringBuilder sb = new StringBuilder();
+      sb.append("B").append(id);
+      if (id == 0) {
+        sb.append("(Exit)\n");
+        return sb.toString();
+      }
+      sb.append("\n");
+      int index = 0;
+      for (Instruction instruction : instructions) {
+        sb.append(index).append(": ").append(Printer.OPCODES[instruction.opcode]).append("\n");
+        index++;
+      }
+      sb.append("Jumps to: ");
+      successors.forEach(s -> sb.append("B").append(s.id).append(" "));
+      sb.append("\n");
+      return sb.toString();
+    }
+  }
+
+  private static class BytecodeCFGMethodVisitor extends MethodVisitor {
+    Map<Label, Block> blockByLabel = new HashMap<>();
+    private Block currentBlock;
+    private BytecodeCFG cfg;
+
+    BytecodeCFGMethodVisitor(Symbol.MethodSymbol methodSymbol) {
+      super(Opcodes.ASM5);
+      cfg = new BytecodeCFG(methodSymbol);
+      currentBlock = new Block(cfg);
```

Comment:
this block is not added to the `cfg.blocks`, so first block is lost. You can see this if you add something before the first `if` in the test, test will continue to work even though there should be additional block.